cpp/src/community/egonet.cu

```diff
@@ -63,56 +68,106 @@ extract(
 {
   auto v           = csr_view.get_number_of_vertices();
   auto e           = csr_view.get_number_of_edges();
-  auto stream      = handle.get_stream();
+  auto user_stream = handle.get_stream_view();
   float avg_degree = e / v;
   rmm::device_vector<size_t> neighbors_offsets(n_subgraphs + 1);
   rmm::device_vector<vertex_t> neighbors;
 
-  // It is the right thing to accept device memory for source_vertex
-  // FIXME consider adding a device API to BFS (ie. accept source on the device)
   std::vector<vertex_t> h_source_vertex(n_subgraphs);
-  raft::update_host(&h_source_vertex[0], source_vertex, n_subgraphs, stream);
+  std::vector<size_t> neighbors_offsets_h(n_subgraphs + 1);
+
+  raft::update_host(&h_source_vertex[0], source_vertex, n_subgraphs, user_stream.value());
```

Comment:
I think you need to call ```user_stream.synchronize()```.

There is no stream synchronization between here and bfs call where h_source_vertex[i] is used; so no guarantee that h_source_vertex[i] will have a valid value.

Reply:
Right, I think it was passing because the user stream was the blocking default stream but I think you're correct that it is unsafe in other cases.

Reply:
f81a2b4